### PR TITLE
Document `at-rule-property-required-list`  difference between properties and descriptors

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

This change was produced by the discussion on <https://github.com/stylelint/stylelint/issues/8148#issuecomment-2531031990>.

> Is there anything in the PR that needs further explanation?

We probably should have named the `at-rule-property-required-list` as `at-rule-descriptor-required-list` because declarations in at-rules are semantically descriptors rather than properties.

See also <https://drafts.csswg.org/css-syntax/#declaration>.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@jeddy3 Do you think we should change the rule's message as well?

https://github.com/stylelint/stylelint/blob/8fe37ab4dae9b35a3aec25bdac120ea1593313d4/lib/rules/at-rule-property-required-list/index.mjs#L12

```diff
-Expected descriptor "${descriptor}" for at-rule "${atRule}"
+Expected descriptor "${descriptor}" for at-rule "${atRule}"
```

I'm a bit concerned about some impact on users. At least, we probably should make the change with another pull request.

File: lib/rules/at-rule-property-required-list/README.md [hunk 1]
```diff
@@ -1,19 +1,19 @@
 # at-rule-property-required-list
 
-Specify a list of required properties for an at-rule.
+Specify a list of required descriptors (previously labelled properties) for an at-rule.
```
Comment (jeddy3) [inline lib/rules/at-rule-property-required-list/README.md:3]:
```suggestion
Specify a list of required descriptors (previously labelled as properties) for an at-rule.
```

**Review by jeddy3 — COMMENTED** [review]
I've left one minor suggestion, otherwise LGTM.

**Comment by jeddy3** [comment]
> Do you think we should change the rule's message as well?

If it doesn't affect our public API, it should be fine to change the message as a fix.

**Comment by Mouvedia** [comment]
> We probably should have named the `at-rule-property-required-list` as `at-rule-descriptor-required-list` because declarations in at-rules are semantically descriptors rather than properties.

Adding an alias is pretty easy.
I am not talking about deprecating the rule though; it would be too disrupting in my opinion.

**Comment by ybiquitous** [comment]
Um... To be honest, I'm not sure if aliasing is easy. It would increase code, and more it might rather produce confusion. 🤔

**Review by Mouvedia — APPROVED** [review]

**Pull request merged**